core/metacling/test/TClingTests.cxx

```diff
@@ -93,52 +95,54 @@ TEST_F(TClingTests, MakeInterpreterValue)
    EXPECT_THAT(v->ToString(), testing::HasSubstr("void my_func_to_print"));
 }
 
-static std::string MakeLibNamePlatformIndependent(llvm::StringRef libName)
+static std::string MakeLibNamePlatformIndependent(const std::string &libName)
 {
    if (libName.empty())
       return {};
-   EXPECT_TRUE(libName.startswith("lib"));
-   EXPECT_TRUE(llvm::sys::path::has_extension(libName));
-   libName.consume_front("lib");
+   EXPECT_EQ(libName.compare(0, 3, "lib"), 0);
+   EXPECT_NE(libName.find('.'), std::string::npos);
    // Remove the extension.
-   return libName.substr(0, libName.find_first_of('.')).str();
+   std::string ret = libName.substr(3, libName.find('.') - 3);
+   return ret;
 }
 
 // Check if the heavily used interface in TCling::AutoLoad returns consistent
 // results.
 TEST_F(TClingTests, GetClassSharedLibs)
 {
    // Shortens the invocation.
-   auto GetLibs = [](const char *cls) -> const char * {
-      return gInterpreter->GetClassSharedLibs(cls);
+   auto GetLibs = [](const char *cls) -> std::string {
+      if (const char *val = gInterpreter->GetClassSharedLibs(cls))
+         return val;
+      return "";
    };
 
-   llvm::StringRef lib = GetLibs("TLorentzVector");
-   EXPECT_STREQ("Physics", MakeLibNamePlatformIndependent(lib).c_str());
+   std::string lib = GetLibs("TLorentzVector");
+   EXPECT_EQ("Physics", MakeLibNamePlatformIndependent(lib));
 
    // FIXME: This should return GenVector. The default args of the LorentzVector
    // are shadowed by Vector4Dfwd.h.
    lib = GetLibs("ROOT::Math::LorentzVector");
-   EXPECT_STREQ("", MakeLibNamePlatformIndependent(lib).c_str());
+   EXPECT_EQ("", MakeLibNamePlatformIndependent(lib));
 
    lib = GetLibs("ROOT::Math::PxPyPzE4D<float>");
-   EXPECT_STREQ("GenVector", MakeLibNamePlatformIndependent(lib).c_str());
+   EXPECT_EQ("GenVector", MakeLibNamePlatformIndependent(lib));
 
    // FIXME: We should probably resolve again to GenVector as it contains the
    // template pattern.
    lib = GetLibs("ROOT::Math::PxPyPzE4D<int>");
-   EXPECT_STREQ("", MakeLibNamePlatformIndependent(lib).c_str());
+   EXPECT_EQ("", MakeLibNamePlatformIndependent(lib));
 
    lib = GetLibs("vector<ROOT::Math::LorentzVector<ROOT::Math::PxPyPzE4D<double> > >");
-   EXPECT_STREQ("GenVector", MakeLibNamePlatformIndependent(lib).c_str());
+   EXPECT_EQ("GenVector", MakeLibNamePlatformIndependent(lib));
```

Comment:
Why the change from _STREQ to _EQ?

Reply:
because the new return type is std::string :)

Reply:
It returns a `std::string` anyway?